examples/python/trigger-pagerduty-incident/handler/handler.py

```diff
@@ -0,0 +1,179 @@
+#
+PAGERDUTY_API_PATH='https://events.pagerduty.com/v2/enqueue'
+PD_CONFIG='/var/openfaas/secrets/pdconfig'
+class FaaSResponse:
+    """
+    FaaSResponse is a helper class to construct a properly formatted message returned by this function.
+    By default, OpenFaaS will marshal this response message as JSON.
+    """    
+    def __init__(self, status, message):
+        """
+        Arguments:
+            status {str} -- the response status code
+            message {str} -- the response message
+        """        
+        self.status=status
+        self.message=message
+
+class Pagerduty:
+    """
+    Pagerduty is a Class used to trigger Events in PagerDuty.
+    """    
+
+    def __init__(self,conn):
+        """
+        Arguments:
+            conn {session} -- connection to PagerDuty REST API
+        """
+        try:
+             with open(PD_CONFIG, 'r') as pdconfigfile:
+                pdconfig = json.load(pdconfigfile)
+                self.routingkey=pdconfig['routing_key']
+                self.event_action=pdconfig['event_action']
+        except OSError as err:
+            res = FaaSResponse('400','Could not read PagerDuty configuration > OSError: {0}'.format(err))
+            print(json.dumps(vars(res)))
+            raise
+        except KeyError as e:
+            res = FaaSResponse('400','Required configuration key not found > KeyError: {0}'.format(err))
+            print(json.dumps(vars(res)))
+            raise
+        self.session=conn
+
+    # PagerDuty REST API implementation        
+    def invoke(self,obj):
+        """
+        Make a rest api call to Pagerduty Events API
+        
+        Arguments:
+            obj {dict} -- Generated API Body for the PagerDuty Events API
+
+        Returns:
+            FaaSResponse -- status code and message
+        """
+        obj['routing_key'] = self.routingkey
+        obj['event_action'] = self.event_action
+        try:
+            resp = self.session.post(PAGERDUTY_API_PATH,json=obj)
+            resp.raise_for_status()
+            print(resp.text)
+            resp_body = json.loads(resp.text)
+            return FaaSResponse('200', 'Successfully {0}ed event action with dedup_key: {1}'.format(self.event_action, resp_body['dedup_key']))
+        except requests.HTTPError as err:
+            return FaaSResponse('500', 'Could not invoke Pagerduty API > HTTPError: {0}'.format(err))
+
+def handle(req):
+    # Validate Event input
+    try:
+        cevent = json.loads(req)
+    except ValueError as err:
+        res = FaaSResponse('400','invalid JSON > ValueError: {0}'.format(err))
+        print(json.dumps(vars(res)))
+        return
+    
+    # Assert that the Event conforms to CloudEvent Spec
+    # For debugging: validate the JSON blob we received - uncomment print statements if needed
+    # print(cevent)
+    try:
+        eventdata = (cevent['data'])
```

Comment:
Is this required and what does it do? (use of tuple)

Reply:
updated. didn't know (variable) had a special meaning in python